### PR TITLE
Fix j1939 tests

### PR DESCRIPTION
**Pull request by battjt**
No significant bugs fixed.  Tests were fixed and the interface was cleaned up a bit in places.

### CODE REVIEW

File: src/org/etools/j1939_84/J1939_84.java [hunk 1]
```diff
@@ -59,22 +59,13 @@ public static Logger getLogger() {
...
-    }
-
     /**
      * Returns true if the application is under test
      *
      * @return true if it's being tested
      */
     public static final boolean isTesting() {
-        return System.getProperty(TESTING_PROPERTY_NAME, "false").equals("true");
+        return Boolean.getBoolean(TESTING_PROPERTY_NAME);
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/J1939_84.java:68]: This won't work.  You're converting a 'key' to a boolean.

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java [hunk 2]
```diff
@@ -162,8 +148,14 @@ public void close() {
...
      */
     @Override
-    public Stream<Packet> duplicate(Stream<Packet> stream) {
-        return inbound.duplicate(stream);
+    public Stream<Packet> duplicate(Stream<Packet> stream, int time, TimeUnit unit) {
+        return inbound.duplicate(stream, time, unit);
+    }
+
+    public void error(String msg, Throwable e) {
+        // FIXME where do we want error messages to go?
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/bus/j1939/J1939TP.java:156]: Write the errors to the log.

File: src/org/etools/j1939_84/J1939_84.java (hunk 1 above)
Comment (battjt) [reply src/org/etools/j1939_84/J1939_84.java:68]: I think this is he appropriate method of reading boolean system properties.  See [Boolean.html#getBoolean(java.lang.String)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Boolean.html#getBoolean(java.lang.String)
)

Comment (battjt) [reply src/org/etools/j1939_84/J1939_84.java:68]: I'd be OK with 
`return System.getProperty(TESTING_PROPERTY_NAME, "false").equalsIgnoreCase("true")`

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java (hunk 2 above)
Comment (battjt) [reply src/org/etools/j1939_84/bus/j1939/J1939TP.java:156]: "warn" messages and application errors go to the same place? One message is for the manufacturer and the other is for the developers.

File: src/org/etools/j1939_84/J1939_84.java (hunk 1 above)
Comment (gumbelmj) [reply src/org/etools/j1939_84/J1939_84.java:68]: I didn't know that method existed.

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java (hunk 2 above)
Comment (gumbelmj) [reply src/org/etools/j1939_84/bus/j1939/J1939TP.java:156]: If there's an error, I think both want to know.  but in this case, the code was just moved so it doesn't need to be resolved right now.

**Pull request merged**